model_managers/model_manager.py

```diff
@@ -474,6 +610,84 @@ def _enrich_node(nd: dict, neighbour: str):
                     merge_on[key].append(_gen_new_prop_name(n))
         return dct, merge_on
 
+    def create_ct(self, controlled_terminology: dict, identifier='label', order_terms=True, merge_on=None):
+        """
+        Creates :Term nodes and links them to a specified class with a [:HAS_CONTROLLED_TERM] relationship.
+        If order terms is True, an ascending Order property will be assigned to terms in the order they are created
+        accounting for the order of existing terms (if any) and [:NEXT] relationships between terms following this order.
+        For example:
+            With identifier = 'label' and controlled_terminology =
+            {
+                'class1': [{'term_label': 'term1'}, {'term_label': 'term2'}],
+                'class2': [{'term_label': 'term3'}]
+            }
+            3 new term nodes with 'term_label' properties equal to 'term1', 'term2' and 'term3' would be
+            created. The class with 'label' = 'class1' would then be linked by [:HAS_CONTROLLED_TERM]
+            relationships to 'term1' and 'term2' and similarly 'class2' would be linked to 'term3'
+
+        :param controlled_terminology: A dictionary of classes and terms eg: {'class1': [{prop1: value, prop2: value}, ...]}
+        :param identifier: String, property used when identifying classes to assign terms.
+        :param order_terms: Bool, if true order properties and next relationships will be created between terms.
+        :param merge_on: Optional List[string] term properties to merge on to prevent duplication
+        :return: neo4j result object.
+        """
+        ident_props = None
+        missing = self.class_exists(list(controlled_terminology.keys()), identifier)
+        assert not missing, f'Cannot create controlled terminology for nonexistent classes: {missing}'
+
+        if merge_on:
+            ident_props = f'{{`{merge_on[0]}`: term_props["{merge_on[0]}"]'
+            for prop in merge_on[1:]:
+                ident_props += f', `{prop}`: term_props["{prop}"]'
+            ident_props += '}'
+
+        # Create terms
+        q1 = f"""
+        UNWIND KEYS($terminology) as class_label
+        MATCH (class:Class {{{identifier}: class_label}})
+        WITH class, class_label
+        UNWIND $terminology[class_label] as term_props
+        CALL apoc.merge.node(['Term', class.label], {f"{ident_props}, term_props, term_props" if ident_props else 'term_props, {}, {}'}) YIELD node as term
```

Comment:
I would change it to
```
CALL apoc.merge.node(['Term'], {f"{ident_props}, term_props, term_props" if ident_props else 'term_props, {}, {}'}) YIELD node
CALL apoc.create.addLabels ([node], [class.label] YIELD node
```
Otherwise you might get into a problem when a Term with given identifying properties exists in the db, but it does not (yet) have a label class.label.